src/session_impl.cpp

```diff
@@ -1384,6 +1394,25 @@ namespace {
 			reopen_outgoing_sockets();
 	}
 
+	std::uint32_t session_impl::get_tracker_key(address const& iface) const
+	{
+		uintptr_t const ses = reinterpret_cast<uintptr_t>(this);
+		hasher h(reinterpret_cast<char const*>(&ses), sizeof(ses));
+		if (iface.is_v4())
+		{
+			auto const b = iface.to_v4().to_bytes();
+			h.update({reinterpret_cast<char const*>(b.data()), b.size()});
```

Comment:
I'm not sure why anyone would do it, but in theory there could be multiple listen sockets with the same address but different ports. To get unique keys for every listen socket in this case the `original_port` would also need to be incorporated into the hash.